aws/resource_aws_secretsmanager_secret_version.go

```diff
@@ -55,8 +65,26 @@ func resourceAwsSecretsManagerSecretVersionCreate(d *schema.ResourceData, meta i
 	secretID := d.Get("secret_id").(string)
 
 	input := &secretsmanager.PutSecretValueInput{
-		SecretId:     aws.String(secretID),
-		SecretString: aws.String(d.Get("secret_string").(string)),
+		SecretId: aws.String(secretID),
+	}
+
+	if v, ok := d.GetOk("secret_string"); ok {
+		input.SecretString = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("secret_binary"); ok {
+		vs := []byte(v.(string))
+
+		if !isBase64Encoded(vs) {
```

Comment:
Not a blocker for merging, but I believe we could enhance this check to actually be done during plan with a `ValidateFunc` on the `secret_binary` attribute:

```go
ValidateFunc: func(v interface{}, name string) (warns []string, errs []error) {
	s := v.(string)
	if !isBase64Encoded([]byte(s)) {
		errs = append(errs, fmt.Errorf("%s: must be base64-encoded", name))
	}
	return
},
```

At some point we can migrate the existing attributes that use this (or a similar) ValidateFunc for ensure Base64 content into its own, easy to use common function like `validateBase64String`. 👍